### PR TITLE
[dedicated] Never show the self-hosted setup modal (by adding option showSetupModal: true)

### PR DESCRIPTION
**Pull request by geropl**
## Description
Add option `showSetupModal` (default: true) to suppress the self-hosted setup-modal. Defaults to `true` for backwards compatibility.

Hopefully we can fade out that thing entirely at some point. :crossed_fingers: 

## Related Issue(s)

Fixes #

## How to test

1. Start [workspace on this branch](https://gitpod-io/#https://github.com/gitpod-io/gitpod/pull/15929) and `kubectl port-forward deployment/server 9000:9000 &`
2. Obtain OTS key: `curl -X POST localhost:9000/admin-user/login-token/create`
3. Use the result to construct your login URL: `https://gpl-test-dedicated.preview.gitpod-dev.com/api/login/ots/admin-user/:yourOtsKey` and visit it
4. You should be logged in as `admin-user` :checkered_flag: (and not blocked by the setup modal)

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all-ci
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-dedicated-emulation
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@geropl: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
   </a>
 </p>
 
+dev
```
Comment (AlexTugarev) [inline README.md:22]: left over

Comment (geropl) [reply README.md:22]: /hold

Comment (geropl) [reply README.md]: Done. Suppressed the build to not block testing.

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 2]
```diff
@@ -408,6 +408,9 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     protected async doUpdateUser(): Promise<void> {
         // execute the check for the setup to be shown until the setup is not required.
         // cf. evaluation of the condition in `checkUser`
+        if (!this.config.showSetupModal) {
+            this.showSetupCondition = { value: false };
```
Comment (AlexTugarev) [inline components/server/src/workspace/gitpod-server-impl.ts:412]: I see, the reason was, that `getUserCount` didn't include the admin account.

**Review by meysholdt — APPROVED** [review]

File: components/server/src/workspace/gitpod-server-impl.ts (hunk 2 above)
Comment (geropl) [reply components/server/src/workspace/gitpod-server-impl.ts:412]: Jup, and it needs to stay that way to keep self-hosted setup working - until we remove it.

**Review by AlexTugarev — APPROVED** [review]

**Review by mrsimonemms — APPROVED** [review]

**Pull request merged**